fortran/micm_core.F90

```diff
@@ -120,16 +124,16 @@ function constructor(config_path, errcode)  result( this )
       end do
       c_config_path(n+1) = c_null_char
 
-      this%ptr = create_micm_c(c_config_path, errcode)
-
-      if (errcode /= 0) then
```

Comment:
A difference here is that we are taking control of the error flow in C++. Whereas what exists now allows for cam-sima to check the result of the error code and control the error flow. I think this makes it confusing to use our c api